### ISSUE TITLE
Issue with fabletools with `c()` fallback in `vec_rbind()`

### ISSUE THREAD

**Issue by lionel-** [r-lib/vctrs#1119]
Since #1111:

```r
library(fabletools)
us_deaths <- as_tsibble(USAccDeaths)
us_deaths_tr <- dplyr::filter(us_deaths, index < tsibble::yearmonth("1978 Jan"))
mbl <- us_deaths_tr %>% model(ets = fable::ETS(value))
fbl <- mbl %>% forecast(h = 12)

fbl %>% dplyr::mutate(`80%` = hilo(value, 80))
#> Error: Problem with `mutate()` input `80%`.
#> ✖ Internal error in `vec_assign()`: `value` should have been recycled to fit `x`.
#> ℹ Input `80%` is `hilo(value, 80)`.
```

FYI @mitchelloharawild, we'll look into this.

**Comment by mitchelloharawild** [comment]
The `hilo` implementation in fabletools is to be replaced by a reimplementation of `hilo` with vctrs in https://github.com/mitchelloharawild/distributional, due for release prior to dplyr v1.0.0 in preparation for the fabletools dplyr update.
Feel free to pursue this if it is a bug, but the issue is currently in the dev packages.

**Comment by mitchelloharawild** [comment]
Minimally reproducible example:
``` r
library(dplyr)
library(distributional)
tibble(dist = dist_normal(1:10)) %>% 
  mutate(hilo(dist))
#> Error: Problem with `mutate()` input `..1`.
#> x Internal error in `vec_assign()`: `value` should have been recycled to fit `x`.
#> ℹ Input `..1` is `hilo(dist)`.
```

<sup>Created on 2020-05-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by lionel-** [comment]
By the way @mitchelloharawild, the problem happened because your distribution class didn't have a self-to-self ptype2 method. I see that you have written methods for the parent class, but these are no longer inherited. Please see the NEWS of 0.3.0 for how to implement methods nowadays. The roxygen syntax has much improved too! I'm happy to take a look at a PR if you'd like.

**Comment by mitchelloharawild** [comment]
Thanks @lionel-.
By self-to-self ptype2 method for the distribution class, are you referring to the one here?
https://github.com/mitchelloharawild/distributional/blob/410982675f168a4ac60728366309ca852f880086/R/distribution.R#L302-L309

I've noticed that I'm missing a self-to-self ptype2 method for the hilo class, which I have now added here https://github.com/mitchelloharawild/distributional/commit/c3db3da0dce3d86c2753ffe0e91006d1c79297b4.

I've dropped the boilerplate code and updated the roxygen syntax in https://github.com/mitchelloharawild/distributional/pull/22, if you'd like to briefly check these changes. It seems to work fine.
